source/grid_generator.py

```diff
@@ -1,17 +1,12 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
-
-import os
-from os import path
-import glob
-import subprocess
-
+import dolfin as dlfn
 from enum import Enum, auto
```

Comment:
I have a question here. 
Only this inheritance ```class (Enum)``` is defined, and then, ```BoundaryMarkers.left.value``` can be used in ```mark()```
```python
class BoundaryMarkers(Enum):    
      left = auto()
gamma01.mark(facet_marker, BoundaryMarkers.left.value)
```
if I don't define that class, just assign ```auto() ```to ```left``` directly, it can not work.
```python
left = auto()
gamma01.mark(facet_marker, left.value)
```
Why must we inherit ```Enum``` and then ```***.value```can work here

Reply:
> Why must we inherit `Enum` and then `***.value`can work here

Well the purpose of the `auto` method is to automatically assign values to the members of an enumeration. The purpose of the enumeration here is provide an easy and understandable way of marking the boundaries for the user. It is not intended to be used outside of an `Enum`  class.

Quote from the python3 documentation: _An enumeration is a set of symbolic names (members) bound to unique, constant values. Within an enumeration, the members can be compared by identity, and the enumeration itself can be iterated over._